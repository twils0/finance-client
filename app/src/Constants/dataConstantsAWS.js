// eslint-disable-next-line
export const actionTypes = {
  SET_AWS_STATUS: 'DATA_AUTH_AWS_STATUS',
  SET_AWS_USER: 'DATA_AUTH_SET_AWS_USER',
  RESET_AWS_STATE: 'DATA_AUTH_RESET_AWS_STATE',
};
