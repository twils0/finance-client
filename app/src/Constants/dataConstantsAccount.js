export const actionTypes = {
  SET_ACCOUNT_STATUS: 'DATA_ACCOUNT_SET_ACCOUNT_STATUS',
  SET_FIELD: 'DATA_ACCOUNT_SET_FIELD',
  SET_FIELDS: 'DATA_ACCOUNT_SET_FIELDS',
  RESET_ACCOUNT_STATE: 'DATA_ACCOUNT_RESET_ACCOUNT_STATE',
};

export const statusNames = {
  AWS_FIELDS: 'accountStatusAWSFields',
  STRIPE_FIELDS: 'accountStatusStripeFields',
  UPDATE_STRIPE_FIELDS_TOKEN: 'accountStatusUpdateStripeFieldsToken',
  UPDATE_STRIPE_FIELDS_REQUEST: 'accountStatusUpdateStripeFieldsRequest',
  UPDATE_DB_FIELDS: 'accountStatusUpdateDBFields',
  UPDATE_AWS_FIELDS: 'accountStatusUpdateAWSFields',
};

export const fieldNames = {
  NAME: 'accountName',
  EMAIL: 'accountEmail',
  EMAIL_ADDITIONAL: 'accountEmailAdditional',
  PHONE: 'accountPhone',
  NAME_ON_CARD: 'accountNameOnCard',
  PROMO_CODE: 'accountPromoCode',
  PROMO_CODE_VALID: 'accountPromoCodeValid',
  CUSTOMER: 'accountCustomer',
};
