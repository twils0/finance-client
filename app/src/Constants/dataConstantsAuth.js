export const actionTypes = {
  SET_AUTHENTICATED: 'DATA_AUTH_SET_AUTHENTICATED',
  SET_REDIRECT_URL: 'DATA_AUTH_SET_REDIRECT_URL',
  SET_AUTH_STATUS: 'DATA_AUTH_SET_AUTH_STATUS',
  SET_CODE_TYPE: 'DATA_AUTH_SET_CODE_TYPE',
  RESET_AUTH_STATE: 'DATA_AUTH_RESET_AUTH_STATE',
};

export const statusNames = {
  LOGIN: 'authStatusLogin',
  SIGN_OUT_DEVICES: 'authStatusSignOutDevices',
  LOGIN_MFA: 'authStatusLoginMFA',
  SIGN_UP: 'authStatusSignUp',
  VERIFY_PHONE: 'authStatusVerifyPhone',
  VERIFY_PHONE_CODE: 'authStatusVerifyPhoneCode',
  VERIFY_EMAIL: 'authStatusVerifyEmail',
  VERIFY_EMAIL_LINK: 'authStatusVerifyEmailLink',
  FORGOT_PASSWORD: 'authStatusForgotPassword',
  RESET_PASSWORD: 'authStatusResetPassword',
  CHANGE_PASSWORD: 'authStatusChangePassword',
  DELETE_ACCOUNT: 'authStatusDeleteAccount',
  LOGOUT: 'authStatusLogout',
};

export const codeTypeNames = {
  VERIFY_PHONE: 'verifyPhone',
  VERIFY_EMAIL: 'verifyEmail',
  VERIFY_EMAIL_ADDITIONAL: 'verifyEmailAdditional',
};
