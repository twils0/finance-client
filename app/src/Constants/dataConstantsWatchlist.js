export const actionTypes = {
  SET_WATCHLIST_STATUS: 'DATA_WATCHLIST_SECURITIES_STATUS',
  SET_SECURITIES_CURRENT: 'DATA_WATCHLIST_SET_SECURITIES_CURRENT',
  SET_SECURITIES_LIST: 'DATA_WATCHLIST_SET_SECURITIES_LIST',
  SET_SECURITIES_ALL: 'DATA_WATCHLIST_SET_SECURITIES_ALL',
  SET_SECURITY: 'DATA_WATCHLIST_SET_SECURITY',
  SET_SECURITY_DATA: 'DATA_WATCHLIST_SET_SECURITY_DATA',
  SET_SECURITY_SUBSCRIBE: 'DATA_WATCHLIST_SET_SECURITY_SUBSCRIBE',
  RESET_WATCHLIST_STATE: 'DATA_ACCOUNT_RESET_WATCHLIST_STATE',
};

export const codeTypeNames = {
  CONFIRM_PHONE: 'confirmPhone',
  CONFIRM_EMAIL: 'confirmEmail',
};

export const statusNames = {
  SECURITIES: 'watchlistStatusSecurities',
  GET_SECURITY_DATA: 'watchlistStatusGetSecurityData',
  UPDATE_SECURITIES: 'watchlistStatusUpdateSecurities',
};
