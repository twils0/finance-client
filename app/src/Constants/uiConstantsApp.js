export const actionTypes = {
  SET_FONTS_STATUS: 'APP_SET_FONTS_STATUS',
  LOAD_FONTS: 'APP_LOAD_FONTS',
  SET_IMAGE_STATUS: 'APP_SET_IMAGE_STATUS',
  LOAD_IMAGE: 'APP_LOAD_IMAGE',
  SET_STRIPE_STATUS: 'APP_SET_STRIPE_STATUS',
  SET_STRIPE_ELEMENT_LOADED: 'APP_SET_STRIPE_ELEMENT_LOADED',
  LOAD_STRIPE: 'APP_FETCH_STRIPE',
  SET_STRIPE_OBJECT: 'APP_SET_STRIPE_OBJECT',
};

export const motionStatusTypes = {
  IDLE: 'IDLE',
  ACTIVE: 'ACTIVE',
};

export const imageNames = {
  EXAMPLE_HEADER: 'exampleHeader',
};

export const errorMessages = {
  NO_EMAIL: 'Please enter your email address.',
  INVALID_EMAIL: 'Please enter a valid email address.',
  NO_PASSWORD: 'Please enter your password.',
  NO_NEW_PASSWORD: 'Please enter your new password.',
  INVALID_PASSWORD:
    'Please enter a password containing at least 8 characters, one of the following special characters, $@!%*#?&, one letter, and one number.',
  NO_MATCH_PASSWORD: 'This password does not match your password above.',
  NO_NAME_ON_CARD: 'Please enter the name shown on your credit card.',
  NO_STRIPE: 'Please enter your credit card information.',
  INVALID_CARD_NUMBER: 'Please enter a valid credit card number.',
  INCOMPLETE_CARD_NUMBER: 'Please enter your full credit card number.',
  INVALID_CARD_EXPIRATION_DATE: 'Please enter a valid expiration date.',
  INVALID_CARD_CVC: 'Please enter a valid CVC.',
  INVALID_CARD_ZIP_CODE: 'Please enter a valid zip code.',
  NO_NAME: 'Please enter your name.',
  NO_PHONE: 'Please enter your mobile phone number.',
  INVALID_PHONE: 'Please enter a valid mobile phone number.',
  NO_VERIFICATION_CODE_PHONE: 'Please enter the verification code sent to your mobile phone.',
  NO_VERIFICATION_CODE_EMAIL: 'Please enter the verification code sent to your email.',
  NO_CONFIRMATION_CODE_PHONE: 'Please enter the confirmation code sent to your mobile phone.',
  INVALID_VERIFICATION_CODE:
    'Please verify you have entered the correct verification code or request a new code.',
  INVALID_CONFIRMATION_CODE:
    'Please verify you have entered the correct confirmation code or request a new code.',
  EXPIRED_VERIFICATION_CODE: 'This verification code has expired.',
  EXPIRED_CONFIRMATION_CODE: 'This confirmation code has expired.',
  EMAIL_IN_USE:
    'An account using this email already exists. Please login or use a different email.',
  EMAIL_IN_USE_INTERNAL:
    'An account using this email already exists. Please use a different email.',
  EMAIL_NOT_FOUND:
    'This email is not associated with an account. Please enter another email or sign up.',
  PASSWORD_NOT_FOUND: 'This password does not match our records.',
  PROMO_CODE_INVALID: 'Please enter a valid promo code.',
  PROMO_CODE_EXPIRED: 'This promo code is no longer valid.',
  RETRY_SIGN_UP: 'Please click "Cancel" below and login.',
  INVALID_DELETE_ACCOUNT: 'Please type "delete" to confirm you would like to delete your account',
  LIMIT_EXCEEDED:
    'You have reached your login attempt limit. Please allow some time and try again.',
  INTERNAL_ERROR:
    'Please try your request again. If unsuccessful on subsequent attempts, please contact customer service.',
};
