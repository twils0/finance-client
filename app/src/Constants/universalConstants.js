import stripeConfig from '../../../stripe.config.json';
import dbConfig from '../../../db.config.json';
import iexConfig from '../../../iex.config.json';

export const baseURLs = {
  STRIPE: stripeConfig.url,
  DB: dbConfig.url,
  IEX: iexConfig.urls.api,
  IEX_DEV: iexConfig.urls.dev,
  IEX_TERMS: iexConfig.urls.terms,
};

export const URLs = {
  CUSTOMERS: '/customers',
  USERS: '/users',
  EMAILS: '/emails',
  SECURITIES: '/securities',
  SECURITIES_SEARCH: '/securities/search',
  IEX_COMPANY: tickerCusip => `/stock/${tickerCusip}/company`,
  IEX_QUOTE: tickerCusip => `/stock/${tickerCusip}/quote`,
};

export const axiosConfig = {
  STRIPE: {
    baseURL: baseURLs.STRIPE,
    timeout: 30000,
  },
  DB: {
    baseURL: baseURLs.DB,
    timeout: 30000,
  },
  IEX: {
    baseURL: baseURLs.IEX,
    timeout: 30000,
  },
};

export const pathNames = {
  DEFAULT: '*',
  LOGIN: '/login',
  SIGN_UP: '/signup',
  CODE: '/code',
  CODE_VERIFY_EMAIL: '/code/:verificationId?',
  TERMS: '/terms',
  INTERNAL: '/internal',
  WATCHLIST: '/internal/watchlist',
  WATCHLIST_SECURITY_ID: '/internal/watchlist/:securityId?',
  ACCOUNT: '/internal/account',
};

export const requestStatusTypes = {
  IDLE: 'IDLE',
  LOADING: 'LOADING',
  SUCCESS: 'SUCCESS',
  ERROR: 'ERROR',
};
