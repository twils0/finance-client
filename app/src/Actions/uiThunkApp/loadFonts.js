import { createGlobalStyle } from 'styled-components';
import WebFont from 'webfontloader';

import { requestStatusTypes } from '../../Constants/universalConstants';
import { setFontsStatus } from '../uiActionsApp';

const loadFonts = () => (dispatch, getState) => {
  const state = getState();

  if (
    state.ui.app.fonts.status !== requestStatusTypes.LOADING
    || state.ui.app.fonts.status !== requestStatusTypes.SUCCESS
  ) {
    dispatch(setFontsStatus({ status: requestStatusTypes.LOADING }));

    WebFont.load({
      google: {
        families: ['Titillium Web:400,600'],
      },
      active: () => dispatch(setFontsStatus({ status: requestStatusTypes.SUCCESS })),
      fontinactive: () => dispatch(setFontsStatus({ status: requestStatusTypes.ERROR })),
    });

    const GlobalFontStyle = createGlobalStyle`
      @font-face {
        font-family: Titillium Web;
      }
    `;

    return GlobalFontStyle;
  }

  return null;
};

export default loadFonts;
