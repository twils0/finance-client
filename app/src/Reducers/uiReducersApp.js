import { combineReducers } from 'redux';

import stripeConfig from '../../../stripe.config.json';
import { requestStatusTypes } from '../Constants/universalConstants';
import { actionTypes, imageNames } from '../Constants/uiConstantsApp';
import {
  formNames as formNamesSignUp,
  inputNames as inputNamesSignUp,
} from '../Constants/uiConstantsExternal';
import {
  formNames as formNamesAccount,
  inputNames as inputNamesAccount,
} from '../Constants/uiConstantsAccount';

import exampleHeader from '../Images/exampleHeader.png';

export const initialState = {
  fonts: {
    status: requestStatusTypes.IDLE,
  },
  images: {
    list: [...Object.values(imageNames)],
    [imageNames.EXAMPLE_HEADER]: {
      id: imageNames.EXAMPLE_HEADER,
      src: exampleHeader,
      status: requestStatusTypes.IDLE,
    },
  },
  stripe: {
    apiKey: stripeConfig.apiKey,
    status: requestStatusTypes.IDLE,
    elements: {
      [inputNamesSignUp[formNamesSignUp.CARD].STRIPE]: {
        id: inputNamesSignUp[formNamesSignUp.CARD].STRIPE,
        loaded: false,
      },
      [inputNamesAccount[formNamesAccount.BILLING].STRIPE]: {
        id: inputNamesAccount[formNamesAccount.BILLING].STRIPE,
        loaded: false,
      },
    },
    stripeObject: null,
  },
};

export const fonts = (state = initialState.fonts, action) => {
  switch (action.type) {
    case actionTypes.SET_FONTS_STATUS:
      return {
        ...state,
        ...action.payload,
      };
    default:
      return state;
  }
};

export const images = (state = initialState.images, action) => {
  switch (action.type) {
    case actionTypes.SET_IMAGE_STATUS:
      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          ...action.payload,
        },
      };
    default:
      return state;
  }
};

export const stripe = (state = initialState.stripe, action) => {
  switch (action.type) {
    case actionTypes.SET_STRIPE_STATUS:
      return {
        ...state,
        ...action.payload,
      };
    case actionTypes.SET_STRIPE_ELEMENT_LOADED:
      return {
        ...state,
        elements: {
          ...state.elements,
          [action.payload.id]: {
            ...state.elements[action.payload.id],
            ...action.payload,
          },
        },
      };
    case actionTypes.SET_STRIPE_OBJECT:
      return {
        ...state,
        ...action.payload,
      };
    default:
      return state;
  }
};

const uiReducersApp = combineReducers({
  fonts,
  images,
  stripe,
});

export default uiReducersApp;
