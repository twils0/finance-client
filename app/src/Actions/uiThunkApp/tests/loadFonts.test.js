import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { injectGlobal } from 'styled-components';
import WebFont from 'webfontloader';

import { initialState as initialStateApp } from '../../../Reducers/uiReducersApp';
import { requestStatusTypes } from '../../../Constants/universalConstants';
import { actionTypes as actionTypesApp } from '../../../Constants/uiConstantsApp';
import loadFonts from '../loadFonts';

jest.mock('styled-components', () => ({
  injectGlobal: jest.fn(),
}));

const middleware = [thunk];
const mockStore = configureMockStore(middleware);

describe('uiThunkApp', () => {
  afterEach(() => {
    injectGlobal.mockReset();
  });

  describe('loadFonts', () => {
    it('calls the correct actions with the correct payloads, when WebFont returns fontinactive', async () => {
      const store = mockStore({ ui: { app: initialStateApp } });

      const expectedActions = [
        {
          type: actionTypesApp.SET_FONTS_STATUS,
          payload: {
            status: requestStatusTypes.LOADING,
          },
        },
        {
          type: actionTypesApp.SET_FONTS_STATUS,
          payload: {
            status: requestStatusTypes.ERROR,
          },
        },
      ];

      WebFont.load = jest.fn(callback => callback.fontinactive());

      const result = store.dispatch(loadFonts());

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
      expect(result).toEqual(null);
      expect(injectGlobal).toMatchSnapshot();
      expect(WebFont.load).toBeCalled();
    });

    it('calls the correct actions with the correct payloads, when WebFont returns fontinactive', async () => {
      const store = mockStore({ ui: { app: initialStateApp } });

      const expectedActions = [
        {
          type: actionTypesApp.SET_FONTS_STATUS,
          payload: {
            status: requestStatusTypes.LOADING,
          },
        },
        {
          type: actionTypesApp.SET_FONTS_STATUS,
          payload: {
            status: requestStatusTypes.SUCCESS,
          },
        },
      ];

      WebFont.load = jest.fn(callback => callback.active());

      const result = store.dispatch(loadFonts());

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
      expect(result).toEqual(null);
      expect(injectGlobal).toMatchSnapshot();
      expect(WebFont.load).toBeCalled();
    });
  });
});
