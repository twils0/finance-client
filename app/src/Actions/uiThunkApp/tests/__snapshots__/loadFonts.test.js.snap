// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uiThunkApp loadFonts calls the correct actions with the correct payloads, when WebFont returns fontinactive 1`] = `[Function]`;

exports[`uiThunkApp loadFonts calls the correct actions with the correct payloads, when WebFont returns fontinactive 2`] = `[Function]`;
